packages/swap/src/routes/quoteV2.ts

```diff
@@ -0,0 +1,365 @@
+          ingressFee,
+          getInternalAsset(srcChainAsset),
+        );
+      }
+      includedFees.push({
+        type: 'INGRESS',
+        chain: srcChainAsset.chain,
+        asset: srcChainAsset.asset,
+        amount: ingressFee.toString(),
+      });
+      swapInputAmount -= ingressFee;
+      if (swapInputAmount <= 0n) {
+        throw ServiceError.badRequest(`amount is lower than estimated ingress fee (${ingressFee})`);
+      }
+
+      if (query.brokerCommissionBps) {
+        const brokerFee = getPipAmountFromAmount(swapInputAmount, query.brokerCommissionBps);
+        includedFees.push({
+          type: 'BROKER',
+          chain: srcChainAsset.chain,
+          asset: srcChainAsset.asset,
+          amount: brokerFee.toString(),
+        });
+        swapInputAmount -= brokerFee;
+      }
+
+      try {
+        const start = performance.now();
+
+        const bestQuote = await getBestQuote({ ...query, amount: swapInputAmount });
+
+        const lowLiquidityWarning = await checkPriceWarning({
+          srcAsset: getInternalAsset(srcChainAsset),
+          destAsset: getInternalAsset(destChainAsset),
+          srcAmount: swapInputAmount,
+          destAmount: bestQuote.outputAmount,
+        });
+
+        const quoteSwapFees = await calculateIncludedSwapFees(
+          getInternalAsset(srcChainAsset),
+          getInternalAsset(destChainAsset),
+          swapInputAmount,
+          bestQuote.intermediateAmount,
+          bestQuote.outputAmount,
+        );
+        includedFees.push(...quoteSwapFees);
+
+        let egressFee = await getEgressFee(destChainAsset);
+        if (egressFee == null) {
+          throw ServiceError.internalError(
+            `could not determine egress fee for ${getInternalAsset(destChainAsset)}`,
+          );
+        }
+        if (ingressEgressFeeIsGasAssetAmount) {
+          egressFee = await estimateIngressEgressFeeAssetAmount(
+            egressFee,
+            getInternalAsset(destChainAsset),
+          );
+        }
+        egressFee = bigintMin(egressFee, BigInt(bestQuote.outputAmount));
+        includedFees.push({
+          type: 'EGRESS',
+          chain: destChainAsset.chain,
+          asset: destChainAsset.asset,
+          amount: egressFee.toString(),
+        });
+
+        const egressAmount = BigInt(bestQuote.outputAmount) - egressFee;
+
+        const minimumEgressAmount = await getMinimumEgressAmount(destChainAsset);
+
+        if (egressAmount < minimumEgressAmount) {
+          throw ServiceError.badRequest(
+            `egress amount (${egressAmount}) is lower than minimum egress amount (${minimumEgressAmount})`,
+          );
+        }
+
+        const {
+          id = undefined,
+          outputAmount,
+          ...response
+        } = {
+          ...bestQuote,
+          egressAmount: egressAmount.toString(),
+          intermediateAmount: bestQuote.intermediateAmount?.toString(),
+          includedFees: [...includedFees, ...bestQuote.fees],
```

Comment:
think this will include some fees two times now

Reply:
No, the network and liquidity fee is handled by the quoting function.